### PR TITLE
feat: 2-step process for SafeManager transferSAFEOwnership

### PR DESCRIPTION
**Pull request by 0xJabberwock**
Closes HAI-260

### CODE REVIEW

File: test/unit/HaiSafeManager.t.sol [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    vm.startPrank(_safeData.owner);
+    safeManager.transferSAFEOwnership(_safe, _safeData.owner);
+  }
+}
+
+contract Unit_AcceptTransferOwnership is Base {
+  event TransferSAFEOwnership(address indexed _sender, uint256 indexed _safe, address _dst);
+
+  function _mockValues(uint256 _safe, IHaiSafeManager.SAFEData memory _safeData) internal {
+    _mockSAFE(_safe, _safeData);
```
Comment (0xJabberwock) [inline test/unit/HaiSafeManager.t.sol]: `_mockValues()` is never called in this scope.

File: test/mocks/HaiSafeManagerForTest.sol [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.19;
+
+import {HaiSafeManager, EnumerableSet} from '@contracts/proxies/HaiSafeManager.sol';
+import {IHaiSafeManager} from '@interfaces/proxies/IHaiSafeManager.sol';
```
Comment (0xJabberwock) [inline test/mocks/HaiSafeManagerForTest.sol]: `IHaiSafeManager` doesn't seem necessary to import.

File: test/unit/HaiSafeManager.t.sol [hunk 3]
```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.19;
+
+import {HaiSafeManager, IHaiSafeManager, ISAFEEngine} from '@contracts/proxies/HaiSafeManager.sol';
+
+import {HaiSafeManagerForTest} from '@test/mocks/HaiSafeManagerForTest.sol';
+import {IAuthorizable} from '@interfaces/utils/IAuthorizable.sol';
```
Comment (0xJabberwock) [inline test/unit/HaiSafeManager.t.sol]: `IAuthorizable` doesn't seem necessary to import.

File: test/unit/HaiSafeManager.t.sol [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+import {HaiTest, stdStorage, StdStorage} from '@test/utils/HaiTest.t.sol';
+
+abstract contract Base is HaiTest {
+  using stdStorage for StdStorage;
```
Comment (0xJabberwock) [inline test/unit/HaiSafeManager.t.sol]: I'd say `not using stdStorage for StdStorage`. 😆

File: test/unit/HaiSafeManager.t.sol [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+  address deployer = label('deployer');
+  address authorizedAccount = label('authorizedAccount');
+  address user = label('user');
```
Comment (0xJabberwock) [inline test/unit/HaiSafeManager.t.sol]: `authorizedAccount` and `user` weren't used, so they may be removed.

**Review by 0xJabberwock — COMMENTED** [review]
👀

**Comment by xBA5ED** [comment]
Good comments! Should be fixed now

**Review by 0xJabberwock — COMMENTED** [review]
✅

**Review by xBA5ED — APPROVED** [review]

**Review by wei3erHase — APPROVED** [review]
looks good to me! 🚢

**Pull request merged**